ammo-fe/src/components/buttons/RecorderButton/index.tsx

```diff
@@ -53,42 +58,51 @@ const RecorderButton = (): ReactElement => {
     const classes = useStyles();
     const i18n = useI18n();
     const connected = useSelector((state: RootReducer) => state.ws.connected);
-    const [recording, setRecording] = useState(false);
+    const recording = useSelector((state: RootReducer) => state.ui.recording);
+
+    const dispatch = useDispatch();
+    const socket = WS.getSocket('http://localhost:3000');
 
     const toggleRecording = (): void => {
-        if (connected) setRecording((prevState) => !prevState);
+        socket.emit('toggleRecord', (isRecording: boolean) => {
+            dispatch({
+                type: UIReducerActionType.TOGGLE_RECORD,
+                recording: isRecording,
+            });
+        });
```

Comment:
Maybe we should start condidering implemeting actions. If we were to reuse this code somewhere else, it'd be nice to have it written someplace factorized